akka-actor-tests/src/test/scala/akka/actor/ActorSystemDispatchersSpec.scala

```diff
@@ -0,0 +1,132 @@
+    }
+
+    def reportFailure(t: Throwable): Unit = {
+      testActor ! "failed"
+      underlying.reportFailure(t)
+    }
+  }
+
+}
+
+class ActorSystemDispatchersSpec extends AkkaSpec with ImplicitSender {
+
+  import ActorSystemDispatchersSpec._
+
+  "The ActorSystem" must {
+
+    "work with a passed in ExecutionContext" in {
+      val ecProbe = TestProbe()
+      val ec = new SnitchingExecutionContext(ecProbe.ref, ExecutionContexts.global())
+
+      val system2 = ActorSystem(name = "ActorSystemDispatchersSpec-passed-in-ec", defaultExecutionContext = Some(ec))
+
+      try {
+        val ref = system2.actorOf(Props(new Actor {
+          def receive = {
+            case "ping" => sender() ! "pong"
+          }
+        }))
+
+        val probe = TestProbe()
+
+        ref.tell("ping", probe.ref)
+
+        ecProbe.expectMsg(1.second, "called")
+        probe.expectMsg(1.second, "pong")
+      } finally {
+        shutdown(system2)
+      }
+    }
+
+    "not use passed in ExecutionContext if executor is configured" in {
+      val ecProbe = TestProbe()
+      val ec = new SnitchingExecutionContext(ecProbe.ref, ExecutionContexts.global())
+
+      val config = ConfigFactory.parseString("akka.actor.default-dispatcher.executor = \"fork-join-executor\"")
+      val system2 = ActorSystem(
+        name = "ActorSystemDispatchersSpec-ec-configured",
+        config = Some(config),
+        defaultExecutionContext = Some(ec))
+
+      try {
+        val ref = system2.actorOf(TestActors.echoActorProps)
+        val probe = TestProbe()
+
+        ref.tell("ping", probe.ref)
+
+        ecProbe.expectNoMessage(200.millis)
+        probe.expectMsg(1.second, "ping")
+      } finally {
+        shutdown(system2)
+      }
+    }
+
+    def userGuardianDispatcher(system: ActorSystem): String = {
+      val impl = system.asInstanceOf[ActorSystemImpl]
+      impl.guardian.asInstanceOf[ActorRefWithCell].underlying.asInstanceOf[ActorCell].dispatcher.id
+    }
+
+    "provide a single place to override the internal dispatcher" in {
+      val sys = ActorSystem(
+        "ActorSystemDispatchersSpec-override-internal-disp",
+        ConfigFactory.parseString("""
+            akka.actor.internal-dispatcher = akka.actor.default-dispatcher
+          """))
+      try {
+        // that the user guardian runs on the overriden dispatcher instead of internal
+        // isn't really a guarantee any internal actor has been made running on the right one
+        // but it's better than no test coverage at all
+        userGuardianDispatcher(sys) should ===("akka.actor.default-dispatcher")
+      } finally {
+        shutdown(sys)
+      }
+    }
+
+    "provide internal dispatcher instance through BootstrapSetup" in {
+      pending // Not sure how important this is and quite tricky to implement, deferring
+      // would be something like
+      // ActorSystem("name", BootstrapSetup().withCustomInternalExecutionContext(myEc))
+    }
```

Comment:
I just realised you could actually achieve something like this, by providing a custom executor like in the first test in this class, and then configure the internal dispatcher to use the default-executor.